### ISSUE TITLE
Quality of Point Clouds

### ISSUE THREAD

**Issue by HungrySkeleton** [iziamtso/P3D#1]
Hello I was wondering what the quality of the point clouds have to be to successfully have the algorithms work. I have tried to pass this simple point cloud of a tomato plant leaf and stems and it appears the software is unable to separate the leaves and stems. I was wondering if there is some threshold that must be met here is an example of the point cloud I am putting into the program. For some context I am using the azure kinect to generate a point cloud then use the all_plants,pb to carry out my classification.
<img width="806" alt="Capture" src="https://user-images.githubusercontent.com/28448412/86834708-bf791b80-c060-11ea-8caf-32f4e30946c5.PNG">

**Comment by iziamtso** [comment]
Hello @HungrySkeleton. Yes, you are correct density of the point cloud is important. The included TensorFlow models were trained on particular density and error. We have used high precision arm laser scanner. (Don't remember exact numbers) I am pretty sure our resolution is LOT higher then kinnect but if you have a lot of your kinnect data and you dont mind label some of it you can train your own TensorFlow model and then simply provide it to the P3D in the menu.  Let me know if this helps.

P.S. In your image I don't see much of "branch" points. Are only interested in separating that little petiole part?
I can give more info on how to train your own model if you want to go that route.

**Comment by sudosurf** [comment]
Thank you for the recent replys here @iziamtso. I'm also interested in training my own model. Did you use an existing tool for labeling the point clouds?

**Comment by iziamtso** [comment]
For labeling you can use CloudCompare tool. It is free and open source. The way I did, I cut different organs of the plant into different point clouds and saved them. Next when I was training I would read all the point cloud files and give them different classes based on the organs I cut into. Basically bring them back into one file again augmented with class markers. Does that make sense to you?

**Comment by HungrySkeleton** [comment]
Hello @iziamtso 

An update I don't have access to a high quality laser scanner to produce high density point clouds. I'm a bit new to tensor flow is it possible you could point me to some resources on how exactly you would train a tensor flow model. Or is it abstracted enough through P3D that all I need to do is just feed in training data (labeled point cloud regions) to the application and it will update the underlying tensor flow model?

I look forward to the next publication hopefully the code goes published as well. I primarily intend this to capture images of tomato / seedlings.
